File: codewars/6-kyu/simple-fun-395---fibonacci-digit-sequence/solution.py
# https://www.codewars.com/kata/5bc555bb62a4cec849000047

# solution 1: 493ms
def find(a, b, n):
    s = str(a) + str(b)

    if n > 20:
        n = n%20 + 20
    
    while len(s) <= n:
        ch = int(s[-1]) + int(s[-2])
        s = s + str(ch)
    
    return int(s[n])

File: codewars/6-kyu/simple-fun-395---fibonacci-digit-sequence/solution.js
// https://www.codewars.com/kata/5bc555bb62a4cec849000047

// solution 1: 1197ms
function find(a,b,n){
    let res = a + '' + b;
    n = +String(n).slice(-4);
    while(res.length <= n)
        res += +res[res.length-2] + +res[res.length-1];
    return +res[n]
}